src/Service/RefundService.php

```diff
@@ -94,7 +94,7 @@ public function doPaymentRefund($transactionId, $amount = null)
                     'amount' => [
                         'currency' => (string)$payment->amount->currency,
                         'value' => (string)TextFormatUtility::formatNumber(
-                            (float)$payment->settlementAmount->value - (float)$payment->amountRefunded->value,
+                            (float)$payment->settlementAmount->value - (float)RefundUtility::getRefundedAmount(iterator_to_array($payment->refunds())),
```

Comment:
Maybe this line 
`(float)$payment->settlementAmount->value - (float)RefundUtility::getRefundedAmount(iterator_to_array($payment->refunds()))`
could be added as getRefundableAmount in RefundUtility and Unit tested

Reply:
go ahead!